### PR TITLE
Reenable substrate api sidecar

### LINKED ISSUE #5106: Reenable substrate api sidecar
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by mariopino**
Closes #5106

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by Bitcoinera** [comment]
So we just try to do this again? Should we include it in the next release?

**Comment by mariopino** [comment]
> So we just try to do this again? Should we include it in the next release?



> So we just try to do this again? Should we include it in the next release?

Nop yet, I'm still testing it

**Comment by mariopino** [comment]
Tested! ready to merge and test in staging ...

File: api/lib/reducers/polkadotV0-reducers.js [hunk 1]
```diff
@@ -527,12 +510,9 @@ function networkAccountReducer(address, account, store) {
...
   }
   return {
-    name:
-      account && account.identity && account.identity.display
-        ? account.identity.display
-        : '',
+    name: account.value ? hexToString(account.value.info.display.Raw) : '',
     address,
-    picture: account ? account.twitter : ''
+    picture: '' // TODO: we need to get images from twitter account
```
Comment (jbibla) [inline api/lib/reducers/polkadotV0-reducers.js:515]: can this be done now?

Comment (mariopino) [reply api/lib/reducers/polkadotV0-reducers.js:515]: Yep, but I would open a different PR for that. Claim rewards notification depends on this to be completed.

File: api/lib/source/polkadotV0-source.js [hunk 2]
```diff
@@ -24,22 +29,61 @@ class polkadotAPI {
...
+    // manually set cache to checking it
+    this.cache = new InMemoryLRUCache()
+    this.httpCache = new HTTPCache(this.cache, this.httpFetch)
+  }
+
   setReducers() {
     this.reducers = require('../reducers/polkadotV0-reducers')
   }
 
+  async getRetry(url, intent = 0) {
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:43]: this is the same code as in Cosmos. let's refactor

File: api/lib/source/polkadotV0-source.js [hunk 3]
```diff
@@ -24,22 +29,61 @@ class polkadotAPI {
...
 
-  async getNetworkAccountInfo(address, api) {
+  async getNetworkAccountInfo(address) {
     if (typeof address === `object`) address = address.toHuman()
     if (this.store.identities[address]) return this.store.identities[address]
+    // TODO: We are not handling sub-identities
     const accountInfo = !this.store.validators[address]
-      ? await api.derive.accounts.info(address)
+      ? await this.query(
+          `${this.baseURL}/pallets/identity/storage/identityOf?key1=${address}`
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:85]: I don't think you need to specify baseURL if it is specified on the RESTDataSource. See Cosmos

File: api/lib/source/polkadotV0-source.js [hunk 4]
```diff
@@ -224,44 +261,54 @@ class polkadotAPI {
...
     // -> Reserved balance (due to identity set) can not be used for anything
     // See https://wiki.polkadot.network/docs/en/build-protocol-info#free-vs-reserved-vs-locked-vs-vesting-balance
-    const { free, reserved, feeFrozen } = account.data.toJSON()
+    const balanceInfo = await this.query(
+      `${this.baseURL}/accounts/${address}/balance-info`
+    )
+
+    // we need addressRole, as /accounts/:address/staking-info
+    // query throws an error if address is not a stash
+    const addressRole = this.getAddressRole(address)
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:292]:
```suggestion
    const addressRole = await this.getAddressRole(address)
```

File: api/lib/source/polkadotV0-source.js (hunk 2 above)
Comment (mariopino) [reply api/lib/source/polkadotV0-source.js:43]: I tried with no luck, you need access to `RESTDataSource` methods in the function and that is not possible outside the source class.

Comment (faboweb) [reply api/lib/source/polkadotV0-source.js:43]: done and pushed. can you check?

**Pull request merged**